exporters/otlp-http/trace/src/main/java/io/opentelemetry/exporter/otlp/http/trace/OtlpHttpSpanExporter.java

```diff
@@ -0,0 +1,222 @@
+  public CompletableResultCode export(Collection<SpanData> spans) {
+    spansSeen.add(spans.size());
+    ExportTraceServiceRequest exportTraceServiceRequest =
+        ExportTraceServiceRequest.newBuilder()
+            .addAllResourceSpans(SpanAdapter.toProtoResourceSpans(spans))
+            .build();
+
+    Request.Builder requestBuilder = new Request.Builder().url(endpoint);
+    if (headers != null) {
+      requestBuilder.headers(headers);
+    }
+    RequestBody requestBody =
+        RequestBody.create(
+            exportTraceServiceRequest.toByteArray(), MediaType.parse("application/x-protobuf"));
+    if (isCompressionEnabled) {
+      requestBuilder.addHeader("Content-Encoding", "gzip");
+      requestBuilder.post(gzipRequestBody(requestBody));
+    } else {
+      requestBuilder.post(requestBody);
+    }
+
+    CompletableResultCode result = new CompletableResultCode();
+
+    client
+        .newCall(requestBuilder.build())
+        .enqueue(
+            new Callback() {
+              @Override
+              public void onFailure(Call call, IOException e) {
+                spansExportedFailure.add(spans.size());
+                result.fail();
+                logger.log(
+                    Level.SEVERE,
+                    "Failed to export spans. The request could not be executed. Full error message: "
+                        + e.getMessage());
+              }
+
+              @Override
+              public void onResponse(Call call, Response response) {
+                if (response.isSuccessful()) {
+                  spansExportedSuccess.add(spans.size());
+                  result.succeed();
+                  return;
+                }
+
+                spansExportedFailure.add(spans.size());
+                int code = response.code();
+
+                Status status = extractErrorStatus(response);
+
+                logger.log(
+                    Level.WARNING,
+                    "Failed to export spans. Server responded with code "
+                        + code
+                        + ". Error message: "
+                        + status.getMessage());
+                result.fail();
+              }
+            });
+
+    return result;
+  }
+
+  private static RequestBody gzipRequestBody(RequestBody requestBody) {
+    return new RequestBody() {
+      @Override
+      public MediaType contentType() {
+        return requestBody.contentType();
+      }
+
+      @Override
+      public long contentLength() {
+        return -1;
+      }
+
+      @Override
+      public void writeTo(BufferedSink bufferedSink) throws IOException {
+        BufferedSink gzipSink = Okio.buffer(new GzipSink(bufferedSink));
+        requestBody.writeTo(gzipSink);
+        gzipSink.close();
+      }
+    };
+  }
+
+  private static Status extractErrorStatus(Response response) {
+    try {
+      ResponseBody responseBody = response.body();
+      if (responseBody == null) {
+        return Status.newBuilder()
```

Comment:
I think we need to pick a non-zero code for this.

Reply:
The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md#otlphttp) says:
> This specification does not use Status.code field and the server MAY omit Status.code field. The clients are not expected to alter their behavior based on Status.code field but MAY record it for troubleshooting purposes.

The status code field isn't currently recorded anywhere for troubleshooting purposes. Still, probably best to to set it in case future maintainers decide to use it. I'll go with `UNKNOWN = 2`.